### PR TITLE
*: update PR template with default release note

### PR DESCRIPTION
**Pull request by francis0407**
### What problem does this PR solve?

Problem Summary:
Because we have a relase note check in CI, we need to add a template for it.
### Release note 
- No release note

### CODE REVIEW

**Comment by SunRunAway** [comment]
For templating, we can make a template release note like 
```
- <!-- write release note here -->
```

Then change the ci to recognize this template and fail it.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -43,3 +43,4 @@ Side effects
 - Breaking backward compatibility
 
 ### Release note <!-- bugfixes or new feature need a release note -->
+- No release note
```
Comment (jackysp) [inline .github/pull_request_template.md:46]: It is better to add a blank line above it.

**Comment by francis0407** [comment]
PTAL @bb7133, we might need to update the CI.

**Comment by bb7133** [comment]
> PTAL @bb7133, we might need to update the CI.

Sure, thanks for updating the template!

Please give me a little more time to review it.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -43,3 +43,5 @@ Side effects
 - Breaking backward compatibility
 
 ### Release note <!-- bugfixes or new feature need a release note -->
+
+- <!-- write release note here, `No release note` if there's no release note -->
```
Comment (bb7133) [inline .github/pull_request_template.md:47]:
```suggestion
- <!-- Please write a release note here to describe the change you made when it is released to the users of TiDB. If your PR doesn't involve any change to TiDB(like test enhancements, RFC proposals...), you can write `No release note` -->
```

Comment (bb7133) [reply .github/pull_request_template.md:47]: How about this?

**Comment by SunRunAway** [comment]
@bb7133 @zhouqiang-cl Ping

**Pull request merged**